src/Button.stories.tsx

```diff
@@ -59,9 +59,9 @@ export function TheAsProp() {
         },
       }}
     >
-      <Checkbox {...checkbox} as={Button}>
+      <Button as={Checkbox} state={checkbox.state} setState={checkbox.setState}>
```

Comment:
I just see now that `Checkbox` breaks with `Button` because `children` prop, I think a could just change to use `value` prop or create a simple component to use with the `as` prop.

What you guys think about that?

Reply:
Maybe this:

```tsx
  const [checked, toggle] = useReducer((value) => !value, false)

  return (
    <ThemeProvider
      theme={{
        colors: {
          background: '#FFFFFF',
          primary: '#2F323A',
          secondary: '#4F5D75',
        },
      }}
    >
      <Button as="a" target="_blank" onClick={toggle}>
        {checked ? '😄 Happy' : '😞 Sad'}
      </Button>
    </ThemeProvider>
  )
```

I just need to have one example of a new prop that don't exists in `ThemeAwareProps`, like `href`